src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1186,6 +1188,21 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
             }
         }
 
+        private bool CanApplyParseOptionChange(ParseOptions oldOptions, ParseOptions newOptions, Project project)
```

Comment:
This still needs to be moved to the VS workspace layer, as the policy you're making here doesn't apply to all workspaces. (For example, it won't work for miscellaneous files workspaces)